### ISSUE TITLE
How to create dropdown list

### ISSUE THREAD

**Issue by darmadiff** [aVadim483/fast-excel-writer#65]
Hello,

I don'w how to create dropdown list using this excel writer. Like using data validation in excel.

**Comment by aVadim483** [comment]
Do you mean filter with dropdown or something else?

**Comment by darmadiff** [comment]
No, not a filter. But I want to fill a cell using dropdown from named range. 
<img width="279" alt="Screenshot 2024-06-24 at 13 52 04" src="https://github.com/aVadim483/fast-excel-writer/assets/130021702/a35d19d1-9892-46c2-af57-c73e2b80ad63">

Something like this

**Comment by aVadim483** [comment]
Ok, I understand, but now there is no such option. I can't promise I'll do it, but I'll think about it

**Comment by aVadim483** [comment]
This feature was implemented in version 6.
https://github.com/aVadim483/fast-excel-writer/blob/master/docs/07-validation.md#datavalidationdropdownformulas

**Comment by alastudio** [comment]
I have try datavalidation, unfortunately method applyDataValidation was not found.

**Comment by aVadim483** [comment]
Have you updated the library to version 6?

**Comment by alastudio** [comment]
Ok, I have updated to version 6 with datavalidation included. Unfortunately, when I try to implement dropdown datavalidation with writeTo, the output of excel file always getting error. And if excel try to recover the error, current sheet with dropdown validation become blank. Even if as simple as
```
$validation = DataValidation::dropDown(['aaa', 'bbb', 'ccc']);
$dataSheet->writeTo('F1', '')->applyDataValidation($validation);
```

**Comment by alastudio** [comment]
It happens when I create named range in another sheet, and apply dropdown validation to current sheet.

```php
//Instructions sheet
$sheet->writeTo('P2', 'YA');
$sheet->writeTo('P3', 'TIDAK');
//$sheet->addNamedRange('P2:P3', 'YA_TIDAK');
$excel->addNamedRange('Instructions!P2:P3', 'YA_TIDAK');

//Data Sheet
$validation = DataValidation::dropDown('=Instructions!YA_TIDAK');
$sheet->writeTo('I2', '')->applyDataValidation($validation);
```

**Comment by alastudio** [comment]
Hi,
I found why excel file becomes error. dropDown list is ok with different sheet, but excel become error if I added mergecells to the file.

**Comment by aVadim483** [comment]
Please show the code how you merge cells

**Comment by alastudio** [comment]
In your sample code, I just put a simple mergecells like this:

```php
...
$sheet = $excel->sheet('Demo');
$sheet->setColWidthAuto('A');
$sheet->mergeCells('C1:C2');
...
```

And excel output file become error.

**Comment by aVadim483** [comment]
No, the problem is something else. I used the following code:
```php
<?php

require_once __DIR__ . '/vendor/autoload.php';

$excel = \avadim\FastExcelWriter\Excel::create(['Demo', 'Test']);

$sheet = $excel->sheet('Demo');
$sheet->setColWidthAuto('A');
$sheet->mergeCells('C1:C2');

$excel->save(__DIR__ . '/test.xlsx');
```
and the file opens without errors

**Comment by alastudio** [comment]
No, I meant if you combine mergeCells with datavalidation dropdown.

```php
require_once __DIR__ . '/../vendor/autoload.php';
require_once __DIR__ . '/../src/autoload.php';

$outFileName = __DIR__ . '/output/' . basename(__FILE__, '.php') . '.xlsx';

use avadim\FastExcelWriter\DataValidation\DataValidation;
use \avadim\FastExcelWriter\Excel;
use avadim\FastExcelWriter\Style;

$timer = microtime(true);

// Create Excel workbook
$excel = Excel::create(['Demo', 'Data']);

$sheet = $excel->sheet('Data');
// Write data for dropdown
$sheet->writeRow(['ddd', 'eee', 'fff']);
// Write named range for dropdown
$sheet->writeRow(['item3-1', 'item3-2', 'item3-3'])->applyNamedRange('list');

$sheet = $excel->sheet('Demo');
$sheet->setColWidthAuto('A');

$style = [
    Style::BORDER => Style::BORDER_THICK,
    Style::FILL_COLOR => '#fffccc',
];

$validation2 = DataValidation::dropDown('=data!A1:C1');
$validation3 = DataValidation::dropDown('=data!list');

$validation = DataValidation::integer('>', 10)
    ->setPrompt('Number greater than 10')
    ->setError('Please enter correct number greater than 10');
;
$sheet->writeCell('Decimal 1');
$sheet->nextCell()
    ->applyStyle($style)
    ->applyDataValidation($validation)
;

$sheet->skipRow();
$validation = DataValidation::dropDown(['aaa', 'bbb', 'ccc']);
$sheet->writeCell('DropDown List 1');
$sheet->writeCell('')
    ->applyStyle($style)
    ->applyDataValidation($validation)
;

$sheet->nextRow();
$sheet->writeCell('DropDown List 2');
$sheet->writeCell('')
    ->applyBorder(Style::BORDER_THICK)
    ->applyBgColor('#fffccc')
    ->applyDataValidation($validation2)
;

$sheet->nextRow();
$sheet->writeCell('DropDown List 3');
$sheet->writeCell('')
    ->applyBorder(Style::BORDER_THICK)
    ->applyBgColor('#fffccc')
    ->applyDataValidation($validation3)
;

$sheet->mergeCells('C1:C2'); //Wherever you want to add mergeCells

// Save to XLSX-file
$excel->save($outFileName);
```

The output file will become error.

**Comment by aVadim483** [comment]
Yes, I confirm the error. It will be fixed in the next release.

**Comment by aVadim483** [comment]
fixed in v6.1.0
